hal/sparcv8leon/hal.c

```diff
@@ -97,5 +97,5 @@ void _hal_init(void)
 	_hal_platformInit();
 	_hal_cpuInit();
 	_hal_consoleInit();
-	_hal_timerInit(SYSTICK_INTERVAL);
+	_hal_timerInit(0xffffffffu);
```

Comment:
Interval is still `SYSTICK_INTERVAL` - in HP timer case it's value still should be used as a maximum interval period for the wakeup timer

Reply:
The only place where `hal_timerSetWakeup` is called is in `_threads_updateWakeup` and there is a check for the wakeup value: https://github.com/phoenix-rtos/phoenix-rtos-kernel/blob/e2a6b1f764dc01ed672afc2bf539a1fd3e147f1a/proc/threads.c#L418
So IMO there's no need to check for interval in timer code